### PR TITLE
Implementation Register Screen 

### PR DESCRIPTION
**Pull request by JitterKiller**
### Summary

This PR includes several significant updates and improvements to the project:

1. **Refactor InfoScreen and PasswordScreen into RegisterScreen**
   - Combined the functionality of InfoScreen and PasswordScreen into a single RegisterScreen 
     (to align with the new Figma).
   - Deprecated the old InfoScreen and PasswordScreen.

2. **Add tests for RegisterScreen and fix existing tests**
   - Added comprehensive unit tests for the new RegisterScreen.
   - Fixed and updated existing tests to ensure compatibility with the new changes.

3. **Update QuickFixTextFieldCustom component**
   - Enhanced the QuickFixTextFieldCustom component for better usability.
   - Replaced all instances of QuickFixTextField with the updated QuickFixTextFieldCustom across all screens.

4. **Update QuickFixCheckBoxRow to support a second clickable label**
   - Improved QuickFixCheckBoxRow to display and handle a second clickable label, enhancing its functionality.

5. **Refactor typography styles**
   - Updated and refactored typography styles to ensure consistency and improve readability across the application.

### Testing

- Added new tests for RegisterScreen.
- Fixed and updated existing tests to align with the new changes.
- Manually tested all screens to ensure the updated QuickFixTextFieldCustom component works as expected.
- Verified the functionality of the updated QuickFixCheckBoxRow.

### Notes

- Ensure to review the deprecated InfoScreen and PasswordScreen for any remaining dependencies.
- Verify the updated typography styles across different screens for consistency.

### Screens

Capture d’écran 2024-10-17 à 16 04 45 Capture d’écran 2024-10-17 à 16 05 13

### CODE REVIEW

**Comment by JitterKiller** [comment]
This PR is ready to be merged @AnotherMedo

**Comment by AnotherMedo** [comment]
> This PR is ready to be merged @AnotherMedo

Thanks! @JitterKiller will make an in depth review rn 😄

File: app/src/androidTest/java/com/arygm/quickfix/ui/authentication/InfoScreenTest.kt [hunk 1]
```diff
@@ -28,6 +28,7 @@ import org.mockito.kotlin.any
 import org.mockito.kotlin.eq
 import org.mockito.kotlin.whenever
 
+@Deprecated("Theses tests are deprecated", ReplaceWith("RegisterScreenTests"))
```
Comment (AnotherMedo) [inline app/src/androidTest/java/com/arygm/quickfix/ui/authentication/InfoScreenTest.kt:31]: Very pertinent use of the annotation

**Review by AnotherMedo — CHANGES_REQUESTED** [review]
This PR disregards changes merged in #66. Please make sure to update your branch before merging. Apart from that everything looks great!

**Review by Autek — APPROVED** [review]
Great refactor and added feature consolidating the screens and improving component functionality. Nice work overall!

**Review by AnotherMedo — APPROVED** [review]
It's all looking great! amazing job :)

**Review by mohagpt — APPROVED** [review]
This pull request looks great! The changes are well-structured and will provide a solid reference for the rest of the team to refactor their own code in a more efficient and maintainable way.

**Pull request merged**